File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/sqlite/IntegerTypes.scala
package dev.typr.foundationssc.docs.sqlite
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object IntegerTypes:
  // start
  val integerType: SqliteType[Long] = SqliteTypes.integer
  val bigintType: SqliteType[Long] = SqliteTypes.bigint
  val intType: SqliteType[Int] = SqliteTypes.int_
  val smallintType: SqliteType[Short] = SqliteTypes.smallint
  val tinyintType: SqliteType[Byte] = SqliteTypes.tinyint
  // stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlite/IntegerTypes.java
package dev.typr.foundations.docs.sqlite;

import dev.typr.foundations.SqliteType;
import dev.typr.foundations.SqliteTypes;

@SuppressWarnings("unused")
public class IntegerTypes {
  // start
  SqliteType<Long> integerType = SqliteTypes.integer; // INTEGER (canonical, INTEGER affinity)
  SqliteType<Long> bigintType = SqliteTypes.bigint; // BIGINT alias
  SqliteType<Integer> intType = SqliteTypes.int_; // INT (32-bit Java int)
  SqliteType<Short> smallintType = SqliteTypes.smallint;
  SqliteType<Byte> tinyintType = SqliteTypes.tinyint;
  // stop
}
